### PR TITLE
Add `except:` param to `Table#pack`

### PR DESCRIPTION
**Pull request by jaynetics**
see #18

### CODE REVIEW

File: lib/tabulo/table.rb
Comment (jaynetics) [inline lib/tabulo/table.rb:210]: i just changed the types here, i think this does not support String or Integer values (?)

File: spec/table_spec.rb
Comment (jaynetics) [inline spec/table_spec.rb:2564]: this was an exact duplicate of the context ending two lines above, so i merged these two

File: lib/tabulo/table.rb
Comment (matt-harvey) [reply lib/tabulo/table.rb:210]: Good pickup. It does actually support an integer; column labels can be integers, strangely enough. You are correct about it not supporting a string, though.

File: lib/tabulo/table.rb
Comment (matt-harvey) [inline lib/tabulo/table.rb:484]: Perfect.

File: lib/tabulo/table.rb
Comment (matt-harvey) [inline lib/tabulo/table.rb:708]: I don't think there's much value in providing `columns:` with a default value here, since the method is private.

If and when these methods are made public, we can then think about a default value.

File: lib/tabulo/table.rb
Comment (matt-harvey) [inline lib/tabulo/table.rb:725]: I don't think there's much value in providing `columns:` with a default value here, since the method is private.

If and when these methods are made public, we can then think about a default value.

File: spec/table_spec.rb
Comment (matt-harvey) [reply spec/table_spec.rb:2564]: 👍  Good pickup.

File: spec/table_spec.rb
Comment (matt-harvey) [inline spec/table_spec.rb:2797]: Please stick to square brackets for `%w[these kinds of arrays]`, for consistency with the project's preexisting style.

File: spec/table_spec.rb
Comment (matt-harvey) [inline spec/table_spec.rb:2804]: Please stick to square brackets for `%w[these kinds of arrays]`, for consistency with the project's preexisting style.

File: spec/table_spec.rb
Comment (matt-harvey) [inline spec/table_spec.rb:2810]: These tests look pretty good. However for this library, I like to test the various parameters, methods etc. against what I expect the end user to actually see, i.e. the rendered table. See the existing tests for precedent.

I know it can be relatively laborious to construct such tests, however, it gives me a level of confidence in correctness of the behaviour, that is greater than what I feel can be provided with internally focused unit tests alone.

Up to you if you want to write the "rendered table" test/s for this feature; I'd be happy to add some more tests myself after merging.

File: README.md
Comment (matt-harvey) [inline README.md:532]: Minor: "an `except:` arguments" should read "an `except:` argument".

Comment (matt-harvey) [reply README.md:532]: I think the README should mention the fact that `except:` can take an array.

If it's too messy to have to spell out both the array form and the symbol form, then I think it would be fine to make it so that `except:` _only_ takes an array, and not also a symbol, just for the sake of simplifying the API.

If it weren't for Rails's precedent, I would definitely prefer that this only took an array; however, so many Ruby developers are used to Rails's conventions that that constitutes a strong argument for accepting either symbol or array.

I don't have a strong preference either way&mdash;I leave it up to you.

**Review by matt-harvey — COMMENTED** [review]
Great contribution. The core of this implementation of the proposed feature looks good to me.

I have left a few minor comments for your consideration.

I am going to be away on holiday for around a week, without computer access. Please forgive me if I don't take any further action on this PR until the new year. I will certainly look at it again shortly after my return though, with a view to merging and then releasing a new minor version very soon.

Many thanks for your work. I hope you have a happy and healthy Christmas and New Year!

File: lib/tabulo/table.rb
Comment (jaynetics) [reply lib/tabulo/table.rb]: OK, I added Integer back in.

File: lib/tabulo/table.rb
Comment (jaynetics) [reply lib/tabulo/table.rb]: OK, I've removed the default. I'd added it just to be extra nice in case someone used these methods via `send`.

File: lib/tabulo/table.rb
Comment (jaynetics) [reply lib/tabulo/table.rb]: OK, I've removed the default. I'd added it just to be extra nice in case someone used these methods via `send`.

File: spec/table_spec.rb
Comment (jaynetics) [reply spec/table_spec.rb]: Done. I also changed an existing example a few lines above where I had picked up these ugly parentheses 😛

File: spec/table_spec.rb
Comment (jaynetics) [reply spec/table_spec.rb]: Done.

File: spec/table_spec.rb
Comment (jaynetics) [reply spec/table_spec.rb:2820]: Done. It actually wasn't laborious at all, I just ran it with `expect(table.to_s).to eq ''` and copied the diff from the output.

File: README.md
Comment (jaynetics) [reply README.md]: I've stuck with supporting both and updated the README.

**Review by matt-harvey — APPROVED** [review]
LGTM!

**Comment by matt-harvey** [comment]
@jaynetics FYI I did some more testing after merging this to `v2-8` branch and pulling down to my local. I found a couple of bugs which I fixed in https://github.com/matt-harvey/tabulo/commit/b4441acd81e0ee9664661d9ee3325a25c42cc140 and https://github.com/matt-harvey/tabulo/commit/88d89822f189d22bf83b15d04308015da536c8b7. We just needed to make sure that the calculations for the desired total table width took _all_ columns into account, not just the ones passed to the `column:` param of `shrink_to` / `expand_to`. 🙂

**Pull request merged**
